### DISCUSSION TITLE
[Feature] Allow settings that are not defined in the config file to be editable in the GUI

### DISCUSSION THREAD

**Discussion by ImranR98** [immich-app/immich#12408] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Currently, the settings UI and config file are mutually exclusive ways to configure Immich - if you have a config file, all UI settings are disabled.

It would be useful if we could apply certain settings through the config file (the UI for those settings would be disabled), but anything missing/undefined in the config file would still be adjustable in the UI.

My use case for this is for a public-facing instance protected by OAuth. Configuring OAuth and disabling email/password login via a config file is much more secure than doing so via the UI. Because in the latter case, there is a short window of time between the app startup and first-time admin setup (during that time anyone could create an admin account since it's an internet-facing instance). It would be good to keep that the config file for auth settings, while leaving everything else for experimentation which is easier through the UI. I'm sure there are other situations where this feature may be useful.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
We considered this before but decided against it as working out which config source to give priority to, and clearly surfacing to the user which config is coming from where, gets pretty tricky.

**Reply by florianblume** [reply]
Hi, first of all thank you for your amazing work, I really love immich!

But I'd second a feature like the requested one, I have exactly the same setup. I use Authelia for Oauth and want to setup everything before the first startup so that the user doesn't have to do that themselves. Wouldn't a simple flag in the config file be possible that says allow UI edits? They then are the highest authority and overwrite other config values. I don't need to be able to see where a config value come from and thought maybe it's possible to just have this simple cascade of configs where the UI has the last say.

**Reply by tago-SE** [reply]
What about allowing unlocking the Admin UI and a confirmation dialog. This would work really nicely with a reset option to remove all GUI edits. The main benefit for this is that as a developer I'd be able to quickly test a setting using the GUI until it works as I want, export the immich.json and update my config file with the desired changes.
